lib/resty/upstream/healthcheck.lua

```diff
@@ -692,40 +784,41 @@ function _M.prometheus_status_page()
 
     local n = #us
 
-    local bits = new_tab(n * 20, 0)
-    local idx = 1
-
-    bits[idx] = "# HELP nginx_upstream_status_info The running status of nginx upstream"
-    idx = idx + 1
-    bits[idx] = "# TYPE nginx_upstream_status_info gauge"
+    local stats_tab = new_status_table(n)
 
     for i = 1, n do
         local u = us[i]
         local ncheckers = upstream_checker_statuses[u]
         if not ncheckers or ncheckers == 0 then
-            idx = idx + 1
-            bits[idx] = string.format("nginx_upstream_status_info{name=\"%s\",endpoint=\"\",status=\"UNKNOW\",role=\"\"} 1", u)
+            add_upstream_unknown_prometheus_status(stats_tab, u)
             goto continue
         end
 
         local peers, err = get_primary_peers(u)
         if not peers then
-            idx = idx + 1
-            bits[idx] = string.format("nginx_upstream_status_info{name=\"%s\",endpoint=\"\",status=\"DOWN\",role=\"\"} 1", u)
+            add_upstream_down_prometheus_status(stats_tab, u)
         else
+            add_upstream_up_prometheus_status(stats_tab, u)
             local peers, err = get_primary_peers(u)
             if peers then
-                idx = gen_peers_prometheus_status_info(peers, bits, idx, u,"PRIMARY")
+                add_peers_prometheus_info(stats_tab, u, peers, "PRIMARY")
             end
 
             peers, err = get_backup_peers(u)
             if peers then
-                idx = gen_peers_prometheus_status_info(peers, bits, idx, u,"BACKUP")
+                add_peers_prometheus_info(stats_tab, u, peers, "BACKUP")
             end
         end
         ::continue::
     end
-    return concat(bits, "\n")
+
+    local status_lines = {
+        "# HELP nginx_upstream_status_info The running status of nginx upstream",
+        "# TYPE nginx_upstream_status_info gauge",
+        unpack(stats_tab.statuses)
```

Comment:
the unpack can not be jit compile. It is better to cancat the status_tab.statuses first.

Reply:
Will do.